test/crit/crit-test.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+set -x
+
+CRIT=../../crit/bin/crit
+
+function gen_imgs {
+	PID=$(./loop)
+	if ! criu dump -v4 -o dump.log -D ./ -t "$PID"; then
+		echo "Failed to checkpoint process $PID"
+		cat dump.log
+		kill -9 "$PID"
+		exit 1
+	fi
+
+	images_list=$(ls -1 ./*.img)
+	if [ -z "$images_list" ]; then
+		echo "Failed to generate images"
+		exit 1
+	fi
+}
+
+function run_test1 {
+	for x in $images_list
+	do
+		echo "=== $x"
+		if [[ $x == *pages* ]]; then
+			echo "skip"
+			continue
+		fi
+
+		echo "  -- to json"
+		$CRIT decode -i "$x" -o "$x"".json" --pretty || exit $?
+		echo "  -- to img"
+		$CRIT encode -i "$x"".json" -o "$x"".json.img" || exit $?
+		echo "  -- cmp"
+		cmp "$x" "$x"".json.img" || exit $?
+
+		echo "=== done"
+	done
+}
+
+
+function run_test2 {
+	mapfile -t array <<< "$images_list"
+
+	PROTO_IN=${array[0]}
```

Comment:
It looks like here we take the first image from `$images_list`. However, it might be easier to debug if we use a specific image instead (e.g., `inventory.img`).